web/admin/application/library/IvozProvider/Klear/Ghost/CarrierServerStatus.php

```diff
@@ -0,0 +1,108 @@
+    {
+        if (!$model->getId()) {
+            return '<span class="ui-silk inline ui-silk-error" title="No carrier server assigned"></span>';
+        }
+
+        /** @var DataGateway $dataGateway */
+        $dataGateway = \Zend_Registry::get('data_gateway');
+
+        /** @var TrunksLcrGatewayDto $kamTrunksLcrGateway */
+        $kamTrunksLcrGateway = $dataGateway->findOneBy(
+            \Ivoz\Kam\Domain\Model\TrunksLcrGateway\TrunksLcrGateway::class,
+            ["TrunksLcrGateway.carrierServer = '". $model->getId() ."'"]
+        );
+
+        if (!$kamTrunksLcrGateway) {
+            return '<span class="ui-silk inline ui-silk-error" title="No carrier server assigned"></span>';
+        }
+
+        $trunksClient = \Zend_Registry::get('container')->get(
+            \Ivoz\Kam\Domain\Service\TrunksClientInterface::class
+        );
+
+        $dumpGw = $trunksClient->getLcrGatewayInfo(
+            $kamTrunksLcrGateway->getId()
+        );
+
+        $status = $dumpGw['state'] ?? '';
+
+        if ($status === 0) {
+            $response = '<span class="ui-silk inline ui-silk-tick" title="Active"></span>';
+        } else {
+            $response = '<span class="ui-silk inline ui-silk-exclamation" title="Inactive""></span>';
+        }
+
+        return $response;
+    }
+
+    /**
+     * Get Carrier Status
+     * @param \Ivoz\Provider\Domain\Model\Carrier\CarrierDto $model
+     * @return string HTML code to display carrier status
+     * @throws Zend_Exception
+     */
+    public function getCarrierStatusIcon($model)
+    {
+        if (!$model->getId()) {
+            return '<span class="ui-silk inline ui-silk-error" title="No carrier assigned"></span>';
+        }
+
+        /** @var DataGateway $dataGateway */
+        $dataGateway = \Zend_Registry::get('data_gateway');
+
+        /** @var CarrierServerDto $carrierServers */
+        $carrierServers = $dataGateway->findBy(
+            \Ivoz\Provider\Domain\Model\CarrierServer\CarrierServer::class,
+            ["CarrierServer.carrier = '". $model->getId() ."'"]
+        );
+
+        $status = 0;
+        foreach ($carrierServers as $carrierServer) {
+            /** @var TrunksLcrGatewayDto $kamTrunksLcrGateway */
+            $kamTrunksLcrGateway = $dataGateway->findOneBy(
+                \Ivoz\Kam\Domain\Model\TrunksLcrGateway\TrunksLcrGateway::class,
+                ["TrunksLcrGateway.carrierServer = '". $carrierServer->getId() ."'"]
+            );
+
+            if (!$kamTrunksLcrGateway) {
+                return '<span class="ui-silk inline ui-silk-error" title="No carrier server assigned"></span>';
+            }
+
+            $trunksClient = \Zend_Registry::get('container')->get(
+                \Ivoz\Kam\Domain\Service\TrunksClientInterface::class
+            );
+
+            $dumpGw = $trunksClient->getLcrGatewayInfo(
+                $kamTrunksLcrGateway->getId()
+            );
+
+            $status += $dumpGw['state'] ?? 2;
```

Comment:
Don't get this. What is the possible state values and why is adding 2 the default in case of error?

A commentary would help IMO